### DISCUSSION TITLE
Does rqlite do any local caching for read-only nodes?

### DISCUSSION THREAD

**Discussion by acuozzo** [rqlite/rqlite#1183] [Q&A] [answered]
My use-case is best articulated by this old StackExchange post: https://serverfault.com/questions/556928/distributed-file-system-with-local-disk-cache

I have lots of little VMs which all connect back to a DB cluster across the internet. My goal is to make as best use of the resources available to them as possible, including the 8 or so GiB of available disk space they have.

I linked to the SE post above because I was (and still am) considering solving this at the filesystem level.

**Comment by otoolep** [comment, marked as answer]
Have you read the docs on read-only nodes? And on read-consistency?

I presume by "caching" what you really want to do is to serve reads from the read-only node, without that node going all the way back to the leader. If so, just select "none" for consistency, and read-only node will serve the data from the local copy of the SQLite database (which is in-memory) by default.

https://rqlite.io/docs/api/read-consistency/#none

So if you send your query to a local read-only node, using the following form:
```bash
curl -G 'localhost:4001/db/query?level=none&freshness=1s' --data-urlencode 'q=SELECT * FROM foo'
```

The read-only node will serve the data by reading the local SQLite database, also ensuring that the data is not more than 1 second old. If it is old, it will return an error (and then you can automatically redirect to the leader by re-issuing the command with `weak` consistency).

Does this answer your question?

**Reply by acuozzo** [reply]
It does. Thank you!

**Comment by otoolep** [comment]
But the disk and RAM usage of a read-only node will be the same as any other node in the cluster, read-only nodes are about avoiding network round trips, not about saving disk and RAM.
